### DISCUSSION TITLE
Why keep an extra VISUAL mode?

### DISCUSSION THREAD

**Discussion by kajuburfi** [helix-editor/helix#397] [Ideas] [answered]
Why is there a need of another made, namely `visual`?
Why not keep it simple, and allow `<s-[key]>` selections?
In Kakoune, it is like this. And I like this feature in kakoune. Likewise, I like features like `space` mode here, which is not available in Kakoune. 
It may be okay to keep the visual mode. But I would suggest to add `shift` selections.

**Comment by pickfire** [comment, marked as answer]
> Why is there a need of another made, namely visual?

It's actually `extend` but we haven't renamed it yet, it also have some issues which is why it wasn't very much documented.

So instead of doing `shift`, you do `v`.

`<s-[key]>` is very limited, you can't do `c-s-f` to extend to next page. And one of the design choices is we want to avoid is holding down `shift` or `alt`.

> Likewise, I like features like space mode here, which is not available in Kakoune.

A few of the `space` features are based on doom emacs.

See https://github.com/helix-editor/helix/issues/165 and https://github.com/helix-editor/helix/wiki/Keymap-brainstorm for more information behind the design choices.

**Reply by kajuburfi** [reply]
Nice keymap idea. But I like holding Shift key also.. So, I have edited my config file..
Good idea to keep that in the config file.
Thanks!

**Reply by pickfire** [reply]
You could change it to use it in the config. I think we didn't open up those functions but you can do that and send a pull request.

**Reply by kajuburfi** [reply]
But you *did* open those specific functions in the `commands.rs`..
I have set my config file (till now) as such:
```toml
"theme" = "gruvbox"

[keys.normal]
S-left = "extend_char_left"
S-right = "extend_char_right"
S-up = "extend_line_up"
S-down = "extend_line_down"
```

This clearly works.. I read about it in the [docs](https://docs.helix-editor.com)

Is it necessary to open a pull request as it already works?

**Reply by CBenoit** [reply]
No it's all good
Glad to see you could get a kakoune-like mapping :slightly_smiling_face:

**Reply by pickfire** [reply]
If you want you can send a contrib/ for kakoune like settings.

**Reply by kajuburfi** [reply]
Sorry, I don't know what that means...
Can you please explain it for me

**Reply by pickfire** [reply]
As in after you tweak the settings you send a pull request into helix contrib/ for the kakoune like settings.

**Reply by kajuburfi** [reply]
Ok, will do
I'll make a pull request with name: `Another Config`?
I would like to make some other changes too. So, the name.

**Reply by kajuburfi** [reply]
@pickfire I am actually planning to implement this now..
But how to go on with this?
Do we make a separate branch? Or another config file?

Ideas?

**Reply by sudormrfbin** [reply]
Making another config file and adding it to [`contrib/`](https://github.com/helix-editor/helix/tree/master/contrib) would be a good idea. And maybe also mention it in the [keymap docs](https://docs.helix-editor.com/keymap.html).

**Reply by kajuburfi** [reply]
How would the user accept that?

**Reply by sudormrfbin** [reply]
They will have to copy it themselves to their own config.

**Reply by kajuburfi** [reply]
Alright then..
Thanks a lot. 👍 

Will do the same.

**Reply by kajuburfi** [reply]
Is it possible to include a toml file in another toml file?
Eg:
In user's `config.toml` file, 
```
import kak_keys.toml
```

or something like that?

Because it would be daunting to copy the file contents to the config.toml..

**Reply by sudormrfbin** [reply]
TOML doesn't support includes, so copying is pretty much the only thing that can be done.

**Reply by pickfire** [reply]
Copy is actually easy, since you can just copy the whole section in the keymap and paste it in your config.
